c++/src/TypeImpl.cc

```diff
@@ -286,7 +286,9 @@ namespace orc {
                                                               MemoryPool& memoryPool, bool encoded,
                                                               bool useTightNumericVector) const {
     switch (static_cast<int64_t>(kind)) {
-      case BOOLEAN:
+      case BOOLEAN: {
```

Comment:
According to the PR title, do we need to use `useTightNumericVector` variable?

Reply:
I think this PR fixes the bug introduced by `useTightNumericVector` which uses `ByteVectorBatch` for `BOOLEAN` accidentally.

Or should we fix this by supporting `ByteVectorBatch` for `BOOLEAN` type?